resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -70,12 +70,38 @@ public void setClientConfiguration(ClientConfiguration configuration)
       this.processor = configuration;
    }
 
-   @Override
-   public Object getEntity()
-   {
-      abortIfClosed();
-      return super.getEntity();
-   }
+	@Override
+	public synchronized Object getEntity() {
+		abortIfClosed();
+		Object entity = super.getEntity();
+		if (entity != null) {
+			return entity;
+		}
+		//Check if the entity was previously fully consumed
+		try {
+			InputStream inputStream = getEntityStream();
+			if(inputStream.available() == 0){

```

Comment:
So there is actually an issue with `inputStream.available()` returning 0 when response header `Transfer-Encoding: chunked` is set without any mention of the chunk size. In such case `InputStream inputStream = getEntityStream();`returns an instance of `org.jboss.resteasy.client.jaxrs.internal.ClientResponse$InputStreamWrapper` which wraps an intance of `org.apache.http.conn.EofSensorInputStream` that wraps an instance of `org.apache.http.impl.io.ChunkedInputStream`.

And the `available()` method of such stream is like:
```
    @Override
    public int available() throws IOException {
        if (this.in instanceof BufferInfo) {
            final int len = ((BufferInfo) this.in).length();
            return (int) Math.min(len, this.chunkSize - this.pos);
        } else {
            return 0;
        }
    }
```

I did not find out where the header was set yet.

Reply:
I keep on trying to understand why the response header `Transfer-Encoding: chunked` is set and it appears that it is set when there is no `Content-Length` header.
I'm facing the problem using  `Resteasy ` and  `Jetty ` only when doing JSON using  `jackson-jaxrs-json-provider`. When doing XML with  `resteasy-jaxb-provider ` it's OK.
So as usual I tried using  `Jersey/Jetty/jackson-jaxrs-json-provider` and  it worked, the  `Content-Length` header was set instead of `Transfer-Encoding: chunked` without chunk size.
So I tried to figure out how does Resteasy set the  `Content-Length` header but I still don't know. If you have any idea about it?